solr/distribution/build.gradle

```diff
@@ -0,0 +1,137 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.apache.lucene.gradle.Checksum
+
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+
+plugins {
+  id 'signing'
+}
+
+// This project puts together the Solr "distribution", assembling bits and pieces
+// from across the project structure into release artifacts.
+
+ext {
+  releaseDir = file("${buildDir}/release")
+  withSignedArtifacts = { ->
+    def propValue = propertyOrDefault("sign", null)
+    // Allow -Psign to work as a shorthand for -Psign=true
+    return propValue != null && (propValue.isBlank() || Boolean.parseBoolean(propValue))
+  }.call()
+}
+
+// Prepare the "source" distribution artifact.
+apply from: buildscript.sourceFile.toPath().resolveSibling("source-release.gradle")
+
+// Prepare the "binary" distribution artifact.
+apply from: buildscript.sourceFile.toPath().resolveSibling("binary-release.gradle")
+
+// Configure artifact signing.
+apply from: buildscript.sourceFile.toPath().resolveSibling("artifact-signing.gradle")
+
+
+// Set up the HTML-rendered "changes" distribution artifact by linking to documentation's output.
+configurations {
+  changesHtml
+  solrTgz
+  solrTgzSignature
+//  docker
+}
+
+artifacts {
+  solrTgz assembleBinaryTgz
+  solrTgzSignature assembleBinaryTgz // TODO: How to reference the signature file from the Sign task?
+}
+
+dependencies {
+  changesHtml project(path: ":solr:documentation", configuration: "changesHtml")
+//  docker project(path: ':solr:docker', configuration: 'packaging')
+}
+
+// Compute checksums for release archives.
+task computeChecksums(type: Checksum) {
+  algorithm = Checksum.Algorithm.SHA512
+
+  files = objects.fileCollection()
+  [
+      tasks.assembleSourceTgz,
+      tasks.assembleBinaryTgz,
+      tasks.assembleBinaryZip,
+  ].each { dep ->
+    dependsOn dep
+    files += dep.outputs.files
+  }
+
+  outputDir = file("${buildDir}/checksums")
+}
+
+
+task prepareGitRev() {
+  dependsOn ":gitStatus"
+
+  ext.outputFile = file("${buildDir}/.gitrev")
+
+  outputs.file(ext.outputFile)
+  inputs.property("gitrev", provider { -> rootProject.ext.gitRev })
+
+  doFirst {
+    Files.writeString(ext.outputFile.toPath(), rootProject.ext.gitRev, StandardCharsets.UTF_8)
+  }
+}
+
+
+// Assemble everything needed in the release folder structure.
+task assembleRelease(type: Sync) {
+  description "Assemble all Solr artifacts for a release."
+  dependsOn rootProject.getTasksByName("mavenToLocalFolder", true)[0]
+
+  from(configurations.changesHtml, {
+    into "changes"
+  })
+
+  from(rootProject.mavenLocalDir, {
+    into "maven"
```

Comment:
I solved the maven artifacts by copying from the mavenToLocalFolder

Reply:
The directory as a source is inferior to a configuration/ artifact because then the task dependency can't be inferred automatically.

Reply:
I tried defining it as a "maven" configuration with artifacts and dependencies, but could not find the right syntax. I can open a JIRA to improve it later, if you don't have the cycles now.

Reply:
https://issues.apache.org/jira/browse/SOLR-15894